spec/helper.js

```diff
@@ -1,9 +1,15 @@
 'use strict';
+const dns = require('dns');
 const semver = require('semver');
 const CurrentSpecReporter = require('./support/CurrentSpecReporter.js');
 const { SpecReporter } = require('jasmine-spec-reporter');
 const SchemaCache = require('../lib/Adapters/Cache/SchemaCache').default;
 
+// Ensures localhost resolves to ipv4 address first on node v17+
+if (dns.setDefaultResultOrder) {
```

Comment:
It's relevant for every `localhost` usage in the code, and it turns out most of them are in the tests.
Still, there are some `localhost` occurences in `src/ParseServer.js`, `src/PostgresConfigParser.js`, `src/cli/parse-server.js` (although it's only printed values, no actually consumed by Node, so these ones are fine), and `src/Options/Definitions.js`.

So indeed there are implications: on Node v17+, the default `serverUrl` of `http://localhost:${options.port}${options.mountPath}` might fail as it resolves to `::1` (whereas on previous versions of Node it would resolve to `127.0.0.1`) but there's no guarantee that the server is indeed listening on the IPv6 interface.

Anyway, the current code works in Node v17, but users would probably have to replace default `localhost` values with `127.0.0.1` instead, or override manually with `dns.setDefaultResultOrder`.

Reply:
That's interesting, also for our parse-server-example repo and potentially for the Parse JS SDK, which is a dependency of Parse Server.

Would you say it makes sense to simply replace all `localhost` references with `127.0.0.1`, also in comments to not confuse developers?

Reply:
@marvinroger what do you think about my previous comment?

Reply:
@mtrezza sorry, pretty busy at work!

That was what I attempted first, but it was quite a big PR.
But I agree that it would probably be more explicit and less error-prone for users used to `localhost`. I can update the PR in that direction, if you're OK!

Reply:
Let's think this through - I assume in the future IPv6 will replace IPv4, so during the current time of transition, `localhost` may be the more universal reference, while 127.0.0.1 would cause issues if a deployment is indeed listening only to IPv6 and not IPv4 (how likely is that?).

Maybe we need to introduce a new Parse Server option to fully support IPv4-only and IPv6-only deployments? Then again I'd look at where in code (not just comments and docs) is there actually a `localhost` reference and what is it used for.

Reply:
Well, I am not sure IPv6-only deployments is currently a thing, at least in Node.
https://nodejs.org/api/net.html#serverlistenport-host-backlog-callback

> In most operating systems, listening to the unspecified IPv6 address (::) may cause the net.Server to also listen on the unspecified IPv4 address (0.0.0.0).

So I think we're safe with `127.0.0.1`. I pushed changes in this direction, but I can revert if needed.
I won't be able to dig deeper though, in particular with IPv4 and IPv6-only support.

Reply:
Your suggestion sounds good.